protocol/standard-validator.md

```diff
@@ -0,0 +1,153 @@
+
+A non-exhaustive list includes:
+
+- `absolutePrestate`: this value varies for each chain. The [current design](https://github.com/ethereum-optimism/optimism/blob/e62c14e64f08ae3cd82973b41315d5797810569b/packages/contracts-bedrock/src/L1/StandardValidator.sol#L61) accepts this value as an input, so no change is needed.
+- Various roles: the default roles should be hardcoded into the StandardValidator, but calls to
+  `validate()` should optionally override them.
+  - note: The
+    [guardian](https://github.com/ethereum-optimism/optimism/blob/e62c14e64f08ae3cd82973b41315d5797810569b/packages/contracts-bedrock/src/L1/StandardValidator.sol#L138)
+    is not currently checked, and the paused state should not be checked.
+- The Permissionless Dispute Game: [should be
+  optional](https://github.com/ethereum-optimism/optimism/blob/e62c14e64f08ae3cd82973b41315d5797810569b/packages/contracts-bedrock/src/L1/StandardValidator.sol#L368).
+
+A small mock up of the implementation approach to allow overridng specific values, focusing on the
+`guardian` value is below:
+
+```solidity
+contract StandardGuardianValidator {
+  struct ValidationInput {
+    IProxyAdmin proxyAdmin;
+    ISystemConfig sysCfg;
+    uint256 l2ChainID;
+  }
+
+  struct ValidationOverrides {
+    address guardian;
+  }
+
+  address immutable GUARDIAN;
+  ISuperchainConfig immutable SUPERCHAIN_CONFIG;
+
+  constructor(address _guardian, ISuperchainConfig _superchainConfig) public {
+    GUARDIAN = _guardian;
+    SUPERCHAIN_CONFIG = _superchainConfig;
+  }
+
+  function expectedGuardian(ValidationOverrides memory _overrides) public view returns (address) {
+    if (_overrides.guardian != address(0)) {
+      return _overrides.guardian;
+    }
+    return GUARDIAN;
+  }
+
+  function validate(
+    bool _allowFailure
+    ValidationInput memory _input,
+    ValidationOverrides memory _overrides
+  ) public view returns (string memory _errors) {
+    _errors = internalRequire(superchainConfig.guardian() == expectedGuardian(_overrides), "SPRCFG-10", _errors);
+  }
+}
+```
+
+### Post-upgrade 16 work
+
+#### 1. OPCM integration:
+
+The `StandardValidator` should become a [component of the OPCM](https://github.com/ethereum-optimism/optimism/blob/0c986c9b40b8aedfce663421d7074fb856229cde/packages/contracts-bedrock/src/L1/OPContractsManager.sol#L1762-L1768),
+so that the `validate()` function can be called directly on the OPCM without needing to track
+another address.
+
+#### 2. Enhance the StandardValidator's status as source of truth.
+
+We should work to include the StandardValidator (as an OPCM component) into the [Configuration Check](https://github.com/ethereum-optimism/OPerating-manual/blob/e1b305a96a0b60515cc1111a26e73e1973d9c34e/Standard%20Rollup%20Charter.md#configuration-check)
+section of the Standard Charter, and if possible remove the links to the specs and registry which
+are currently listed there.
```

Comment:
TODO: look into how standard-config-params-mainnet is currently used. 
Can we deprecate it?
OR should it be the SoT that is used to deploy the StandardValidator.

Reply:
I have inquired about this with Slipper in discord, and updated the source of truth section to reflect it.